arbitrum/apibackend.go

```diff
@@ -445,20 +446,57 @@ func (a *APIBackend) stateAndHeaderFromHeader(ctx context.Context, header *types
 	}
 	bc := a.BlockChain()
 	stateFor := func(header *types.Header) (*state.StateDB, error) {
```

Comment:
because StateFor is only used internaly and in two places, I think it's safe to have it return a release function and use it.
FindLastAvailableState can also return the release function for latest available state.
Then we'll only set a finalizer in stateAndHeaderFromHeader